src/serde.rs

```diff
@@ -26,10 +26,6 @@ pub fn default_framing_message_based() -> FramingConfig {
     BytesDecoderConfig::new().into()
 }
 
-pub fn default_framing_stream_based() -> FramingConfig {
-    NewlineDelimitedDecoderConfig::new().into()
-}
-
```

Comment:
Can we remove the `default_framing_message_based` a couple of lines above or is that still used in placed related to this?

Reply:
No, this only changed stream-based sources because they're the only ones that actually need framing by default. The `default_framing_message_based` is `BytesDecoder`, which is the semantic equivalent of no framing at all and appropriate for all codecs in message-based sources. I'd personally find it a bit clearer to have these places use `Option<Framer>` instead of defaulting to a framer that doesn't do anything, but that's just my opinion 😄

Reply:
`Option<Framer>` is an interesting idea.

Reply:
> I'd personally find it a bit clearer to have these places use Option<Framer> instead of defaulting to a framer that doesn't do anything 😄

Yeah we can revisit that. Just want to note that if we go for `Option<Framer>`, then there's an additional conditional check in every `Decoder` for the `Option` even when we know there's always a framer. Since that's a pretty hot path, we should benchmark that well before introducing it.

I agree either way that this reasoning is not obvious and should maybe be a comment in the `Decoder` source code somewhere 🤔